### ISSUE TITLE
[BUG] Incorrect format headers returned by performance analyzer

### ISSUE THREAD

**Issue by jainankitk** [opensearch-project/performance-analyzer#203]
**What is the bug?**

The headers returned by PA plugin are incorrect as it returns null key and list as value.

Header validation exception due to null / empty header
```
< HTTP/1.1 500 Internal Server Error
< content-type: application/json; charset=UTF-8
< content-length: 181
<
* Connection #0 to host localhost left intact
{"error":{"root_cause":[{"type":"illegal_state_exception","reason":"Channel is already closed"}],"type":"illegal_state_exception","reason":"Channel is already closed"},"status":500}
```

Date value returned as list due to it being stored as List<List<String>>
```
HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
Date: [Fri, 11 Mar 2022 11:59:36 GMT]
Content-Type: text/plain; charset=UTF-8
Content-Length: 157
```

**Comment by jainankitk** [comment]
> Please don't close this issue until tests have been added on top of #204

The validation of headers is automatically done by netty due to which the agent started throwing errors. Though, I noticed that integration tests were directly using rca on 9600 port instead of going via agent on 9200. Have raised PRs to address that issue - https://github.com/opensearch-project/performance-analyzer/pull/208, https://github.com/opensearch-project/performance-analyzer-rca/pull/177

**Comment by dblock** [comment]
@jainankitk All tests passed before your change in https://github.com/opensearch-project/performance-analyzer/pull/204, and have passed after the same. So there are no tests that catch the bug described here and fixed in #204. Am I missing something?

**Comment by jainankitk** [comment]
@dblock - I guess the test suite does not run the integration tests. Without #208, even the integration tests would not have caught this issue, which is not the case now.
Not sure if we have automated way of running integration test suite, maybe nightly if not with every commit.
